### ISSUE TITLE
Evillimiter only showing upload activity and control, no download

### ISSUE THREAD

**Issue by austrian-oak** [bitbrute/evillimiter#152]
Hello

I am using latest Kali Linux system, i did setup evil-limiter correctly and sound to be working fine. My issue is that it seems like it only controls upload, even monitor and analyze shows only activities in the upload part, but nothing in the download.
![upload](https://github.com/bitbrute/evillimiter/assets/148447653/a4df0852-187c-4835-947f-0e9678780238)

**Comment by austrian-oak** [comment]
Partially solved by allowing NAT over all traffic
iptables -t nat -A POSTROUTING -j MASQUERADE
now i get the upload and download usage correctly when i use "analyze", but the problem still it that bandwidth limits only upload and has no effect on download.
Any suggestions ?

**Comment by austrian-oak** [comment]
solved it finally. 
using wire shark showed that I'm not broadcasting myself properly to router and clients, so i used ettercap the graphical edition to perform ARP poisoning , then i changed the marked packet from PREROUTING  to OUTPUT in the limit.py file in evillimiter root folder, ran setup again and it worked, tested it and fully functional.

Note: make sure ipv4 forward is working, otherwise the clients won't have internet access.

**Comment by austrian-oak** [comment]
Solved

**Comment by austrian-oak** [comment]
i guess as its closed i can elaborate a bit for future reference for those who might face the same issue. Evil limiter in the beginning was working only on uploads, in terms of analysis and control, i tried to see any work done on the marked packets through "iptables -t mangle -n -v -L", and i noticed that when limiting up/download bandwidth, the only change in packets number and byte size was in the postrouting marked packets. i tried many things but it didn't work. so i switched to the windows version of the app and it was behaving exactly the same. i then decided to move on to another application which was netstalker windows version, it wasnt doing any good either, it seemed like there was no real way to control the download bandwidth.
one thing i noticed though, it was by both trial and coincidence, after i ticked the "redirect all" option in the net stalker, evillimiter "windows version started to work as it should. I tried to figure out what was happening, i am new to linux  and i googled my way through. i was about to give up, so i decided to go back to basics, wireshark and the original source code of the app " and here i would like to thank everyone for making access to the source code possible" it started to make sense. i i knew i had to posistion my laptop -which i was using as a hack device- in the middle between the router and the other clients, and i saw that happening when i ticked redirect all in netstalker, i was broadcasting myself properly as intended for this purpose. 
I tried to reflect that many times in the linux version but i couldn't, i tried changing variables in the .py files of evillimiter but it didn't work as i was not doing it properly. i tried to use the usual arp.spoof to broadcast myself properly for everyone in the network , but i found that MitM was doing it better. though the MitM command didn't work for me, i used ettercap to do it, and before using evillimiter in linux, i wiresharked the packets and i saw it was doing great. i then enabled ipv4 forwarding, and changed the "prerouting" command line in "limit.py" to "forward" and added a similar line for "output', i really didn't want to test them individually as i was already tired and glad that it worked.
i wish a more experienced one would be able to embed that in the original code of the app, so that no one needs to do it further.
